packages/grafana-runtime/src/services/EchoSrv.ts

```diff
@@ -25,40 +40,96 @@ export interface EchoMeta {
   timeSinceNavigationStart: number;
 }
 
+/**
+ * Type interface describing echo backends that can be registered to receive of events.
+ *
+ * @public
+ */
 export interface EchoBackend<T extends EchoEvent = any, O = any> {
   options: O;
   supportedEvents: EchoEventType[];
   flush: () => void;
   addEvent: (event: T) => void;
 }
 
+/**
+ * Type interface describing an echo event.
+ *
+ * @public
+ */
 export interface EchoEvent<T extends EchoEventType = any, P = any> {
   type: EchoEventType;
+  /**
+   * Event payload containing event specific data.
+   */
   payload: P;
   meta: EchoMeta;
 }
 
+/**
+ * Supported echo event types that can be sent via the {@link EchoSrv}.
+ *
+ * @public
+ */
 export enum EchoEventType {
   Performance = 'performance',
   MetaAnalytics = 'meta-analytics',
 }
 
+/**
+ * Used to send events to all the registered backends. This should be accessed via the
+ * {@link getEchoSrv()} function. Will, by default, flush events to the backends every
+ * 10s or when the flush function is triggered.
+ *
+ * @public
+ */
 export interface EchoSrv {
+  /**
+   * Call this to flush current events to the echo backends.
+   */
   flush(): void;
+  /**
+   * Add a new echo backend to the list of backends that will receive events.
+   */
   addBackend(backend: EchoBackend): void;
+  /**
+   * Call this to add event that will be sent to the echo backends upon next
+   * flush.
+   *
+   * @param event - Object containing event information.
+   * @param meta - Object that will extend/override the default meta object.
+   */
   addEvent<T extends EchoEvent>(event: Omit<T, 'meta'>, meta?: {}): void;
 }
 
 let singletonInstance: EchoSrv;
 
+/**
+ * Used during startup by Grafana to set the EchoSrv so it is available
+ * via the the {@link getEchoSrv()} to the rest of the application.
+ *
+ * @internal
+ */
 export function setEchoSrv(instance: EchoSrv) {
   singletonInstance = instance;
 }
 
+/**
+ * Used to retrieve the {@link EchoSrv} that can be used to report events to registered
+ * echo backends.
+ *
+ * @public
+ */
 export function getEchoSrv(): EchoSrv {
   return singletonInstance;
 }
 
+/**
+ * Used to register echo backends that will receive Grafana echo events during application
+ * runtime.
+ *
+ * @public
+ */
 export const registerEchoBackend = (backend: EchoBackend) => {
```

Comment:
note: Feels a bit dangerous to give plugins etc. the possibility to listen to all meta events within Grafana. Do we send any sensitive information?

Reply:
Good question, worth aligning with Enterprise team.

Reply:
🤔

1. We're listening to the meta events from inside of Grafana, so we don't need it to be accessible for plugins.
2. The information is personal but not sensitive, it responds to the question "who loaded what panel when and how long time did it take". The possibly sensitive thing is that the `error` field is a free-text with whatever error was returned.

Reply:
1. OK ok, that is good! I don't think, with the current design of the packages, we can limit this. Might be something we revisit after 7.0.

2. Super!

@dprokop what do you think?

Reply:
With the above information I think this should be OK. Worth revisiting as you mentioned @mckn